modules/base/rendering/renderablenodeline.cpp

```diff
@@ -0,0 +1,307 @@
+    using namespace documentation;
+    return {
+        "Renderable Node Line",
+        "base_renderable_renderablenodeline",
+        {
+            {
+                StartNodeInfo.identifier,
+                new StringVerifier,
+                Optional::No,
+                StartNodeInfo.description
+            },
+            {
+                EndNodeInfo.identifier,
+                new StringVerifier,
+                Optional::No,
+                EndNodeInfo.description
+            },
+            {
+                LineColorInfo.identifier,
+                new DoubleVector3Verifier,
+                Optional::Yes,
+                LineColorInfo.description
+            },
+            {
+                LineWidthInfo.identifier,
+                new DoubleVerifier,
+                Optional::Yes,
+                LineWidthInfo.description
+            }
+        }
+    };
+}
+
+RenderableNodeLine::RenderableNodeLine(const ghoul::Dictionary& dictionary)
+    : Renderable(dictionary)
+    , _lineColor(LineColorInfo, glm::vec3(1.f, 1.f, 1.f), glm::vec3(0.f), glm::vec3(1.f))
+    , _lineWidth(LineWidthInfo, 2.f, 1.f, 20.f)
+    , _start(StartNodeInfo, Root)
+    , _end(EndNodeInfo, Root)
+{
+    documentation::testSpecificationAndThrow(
+        Documentation(),
+        dictionary,
+        "RenderableNodeLine"
+    );
+
+    _start = dictionary.value<std::string>(StartNodeInfo.identifier);
+    _end = dictionary.value<std::string>(EndNodeInfo.identifier);
+
+    if (dictionary.hasKey(LineColorInfo.identifier)) {
+        _lineColor = dictionary.value<glm::vec3>(LineColorInfo.identifier);
+    }
+    if (dictionary.hasKey(LineWidthInfo.identifier)) {
+        _lineWidth = float(dictionary.value<double>(LineWidthInfo.identifier));
+    }
+
+    addProperty(_start);
+    addProperty(_end);
+    addProperty(_lineColor);
+    addProperty(_lineWidth);
+    addProperty(_opacity);
+}
+
+void RenderableNodeLine::initializeGL() {
+    _program = BaseModule::ProgramObjectManager.request(
+        ProgramName,
+        []() -> std::unique_ptr<ghoul::opengl::ProgramObject> {
+            return global::renderEngine.buildRenderProgram(
+                ProgramName,
+                absPath("${MODULE_BASE}/shaders/line_vs.glsl"),
+                absPath("${MODULE_BASE}/shaders/line_fs.glsl")
+            );
+        }
+    );
+
+    // Generate
+    glGenVertexArrays(1, &_vaoId);
+    glGenBuffers(1, &_vBufferId);
+
+    bindGL();
+
+    glVertexAttribPointer(_locVertex, 3, GL_FLOAT, GL_FALSE, 3 * sizeof(float), (void*)0);
+    glEnableVertexAttribArray(_locVertex);
+
+    unbindGL();
+}
+
+void RenderableNodeLine::deinitializeGL() {
+
+    glDeleteVertexArrays(1, &_vaoId);
+    _vaoId = 0;
+
+    glDeleteBuffers(1, &_vBufferId);
+    _vBufferId = 0;
+
+    BaseModule::ProgramObjectManager.release(
+        ProgramName,
+        [](ghoul::opengl::ProgramObject* p) {
+        global::renderEngine.removeRenderProgram(p);
```

Comment:
This and the next line are missing an indentation level (thanks Visual Studio for autocorrecting this *all the time*)